### PR TITLE
Persist token limit slider value and expose state setters

### LINKED ISSUE #296: fix(web): Persist token limit slider value across sessions/navigations
## Description
The token limit slider (default 8000) does not persist its value when changed. Navigating away or refreshing the page resets the slider to 8000, and programmatic changes via JavaScript are not reflected in the UI.

## Steps to Reproduce
1. Open the production website
2. Change the slider value to 4000 (via UI drag or JavaScript eval)
3. Refresh the page
4. Observe that the slider resets to 8000
5. Navigate to Help page and back
6. Observe slider reset to 8000

## Expected Behavior
Slider value should be persisted in client-side storage (localStorage/sessionStorage) and restored on page load/navigation.

## Actual Behavior
Slider value is not persisted, and programmatic changes may not update the UI.

## Environment
- Production URL: https://web-eight-ivory-29.vercel.app/
- Browser: Chromium (agent-browser)
- Device: Desktop

## Files Affected
- `web/` (slider component, state persistence)

### PR DESCRIPTION
**Pull request by d-oit**
The token limit slider and other UI configuration settings now persist across page refreshes and navigations. The initial state is loaded from localStorage synchronously within a useEffect hook to ensure a seamless transition from defaults to saved values without hydration errors. Additionally, key state setters are exposed to the global window object to allow for programmatic control of the UI from external JavaScript. A new E2E test suite has been added to verify these changes.

Fixes #296

---
*PR created automatically by Jules for task [16476803667499315823](https://jules.google.com/task/16476803667499315823) started by @d-oit*

### CODE REVIEW

**Comment by d-oit** [comment]
@jules Address all PR review feedback completely and systematically.

Requirements:

* Analyze every review comment, requested change, CI signal, and inline discussion on the PR.
* Implement fixes with production-quality changes only — no temporary workarounds, commented-out code, or partial resolutions.
* Preserve existing behavior unless explicitly required by the feedback.
* Verify there are no regressions, edge-case failures, or broken integrations.

Validation workflow:

1. Re-run the full validation pipeline locally:

   * tests
   * lint
   * typecheck
   * build
   * benchmarks (if applicable)
2. Use real usage / dogfooding flows where possible instead of only unit tests.
3. Verify affected functionality through the actual CLI and end-to-end execution paths.
4. Confirm CI parity with local execution.
5. Inspect git diff carefully for accidental unrelated changes.

Git workflow:

* Commit all changes with clear conventional commit messages.
* Push all commits to the active PR branch.

PR verification (mandatory):

* Use GitHub CLI (`gh`) to inspect the PR after pushing.
* Verify:

  * all required checks are green or running successfully
  * no merge conflicts exist
  * PR state is `mergeable`
  * no unresolved review threads remain
  * branch is up to date with target branch if required

Use commands such as:

* `gh pr view --comments --json reviews,reviewThreads,mergeable,headRefName,statusCheckRollup`
* `gh pr checks`
* `gh pr status`

If the PR is not mergeable:

* identify the blocking issue
* resolve it completely
* re-verify until the PR is mergeable

Finally provide:

* concise summary of implemented fixes
* validation results
* CI/check status
* final mergeability status
* remaining risks or follow-ups (if any)

File: web/tests/e2e/token-persistence.spec.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  }
+}
+
+async function closeSidebarIfMobile(page: Page) {
+  const isMobile = (page.viewportSize()?.width || 0) < 1024;
+  if (isMobile) {
+    await page.evaluate(() => {
+      if ((window as any).__WDR_INTERNAL_STATE_SETTERS__) {
+        (window as any).__WDR_INTERNAL_STATE_SETTERS__.setMobileMenuOpen(false);
```
[the remark this answers is not in the document]
Comment (d-oit) [reply web/tests/e2e/token-persistence.spec.ts:22]: @jules address feedback

File: web/tests/e2e/token-persistence.spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      await expect(backdrop).toBeVisible();
+    }
+  }
+}
+
+async function closeSidebarIfMobile(page: Page) {
+  const isMobile = (page.viewportSize()?.width || 0) < 1024;
+  if (isMobile) {
+    await page.evaluate(() => {
+      if ((window as any).__WDR_INTERNAL_STATE_SETTERS__) {
```
[the remark this answers is not in the document]
Comment (d-oit) [reply web/tests/e2e/token-persistence.spec.ts:21]: @jules address feedback, git commit, git push all changes—verify with gh cli the pr. pr must be mergeable.

File: web/tests/e2e/token-persistence.spec.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  await page.goto("/");
+  await expect(page.getByTestId("app-loaded")).toBeVisible({ timeout: 15000 });
+  await openSidebarIfMobile(page);
+
+  const slider = page.locator('aside input[type="range"]').first();
+  const label = page.locator('aside span.text-accent').first();
+
+  await page.evaluate(() => {
+    if ((window as any).__WDR_INTERNAL_STATE_SETTERS__) {
+      (window as any).__WDR_INTERNAL_STATE_SETTERS__.setMaxChars(25000);
```
[the remark this answers is not in the document]
Comment (d-oit) [reply web/tests/e2e/token-persistence.spec.ts:89]: @jules address feedback, git commit, git push all changes—verify with gh cli the pr. pr must be mergeable.
